src/app/shared/navigation/navigation.component.ts

```diff
@@ -83,12 +91,26 @@ export class NavigationComponent {
   ];
 
   public constructor(
+    private readonly featureStateResolver: FeatureStateResolver,
     private readonly navigationListService: NavigationListService,
     private readonly preferenceService: PreferenceService,
     private readonly navListComponentService: NavigationListComponentService,
     private readonly activatedRoute: ActivatedRoute,
     private readonly timeRangeService: TimeRangeService
   ) {
+    this.subscriptions.add(
+      this.featureStateResolver.getFeatureState(ApplicationFeature.SavedQueries).subscribe(featureState => {
+        if (featureState === FeatureState.Enabled) {
+          this.navItemDefinitions.push({
+            type: NavItemType.Link,
+            label: 'Saved Queries',
+            icon: IconType.Save,
+            matchPaths: ['saved-queries']
+          });
+        }
+      })
+    );
+
```

Comment:
If we can make this generic enough and load the `saved-queries` route metadata from somewhere else, it will help in isolation and readability.

Reply:
Not sure where else to put this metadata. The metadata for all the other routes is defined in the same file too:

https://github.com/razorpay/hypertrace-ui/blob/410d6ed317dbda1505d37d06f05967b6622d6a1a/src/app/shared/navigation/navigation.component.ts#L46-L52

Should I move the metadata for all routes to some kind of constants file and read it from there?

Reply:
Discussed about this [here](https://razorpay.slack.com/archives/C03A6MZF5U4/p1653466465778689)

Reply:
Moved the saved queries metadata to a constants file ✅